frontend/src/metabase/dashboard/components/ClickBehaviorSidebar/LinkOptions/LinkedEntityPicker/LinkedEntityPicker.unit.spec.tsx

```diff
@@ -0,0 +1,382 @@
+          const typedText = "question";
+          userEvent.type(
+            screen.getByPlaceholderText("Search"),
+            `${typedText}{enter}`,
+          );
+
+          expect(
+            await screen.findByText(questionSearchResult.name),
+          ).toBeInTheDocument();
+
+          const call = fetchMock.lastCall("path:/api/search");
+          const urlObject = new URL(checkNotNull(call?.request?.url));
+          expect(urlObject.pathname).toEqual("/api/search");
+
+          expect(urlObject.searchParams.getAll("models")).toEqual([
+            "card",
+            "dataset",
+          ]);
+          expect(
+            fromEntriesWithMultipleValues(urlObject.searchParams.entries()),
+          ).toEqual({
+            models: ["card", "dataset"],
+            q: typedText,
+            filter_items_in_personal_collection: "exclude",
+          });
+        });
+      });
+    });
+
+    describe("dashboard in a personal collection", () => {
+      const dashboardInPersonalCollection = createMockDashboard({
+        collection: PERSONAL_COLLECTION,
+        collection_id: PERSONAL_COLLECTION.id as number,
+      });
+
+      it("should render all collections", async () => {
+        setup({
+          clickBehavior,
+          dashboard: dashboardInPersonalCollection,
+          collectionItems: [questionCollectionItem],
+        });
+
+        expect(
+          screen.getByText("Pick a question to link to"),
+        ).toBeInTheDocument();
+        expect(
+          await screen.findByText(PUBLIC_COLLECTION.name),
+        ).toBeInTheDocument();
+        expect(screen.getByText(PERSONAL_COLLECTION.name)).toBeInTheDocument();
+        expect(
+          await screen.findByText(questionCollectionItem.name),
+        ).toBeInTheDocument();
+      });
+
+      describe("search questions", () => {
+        it("should search questions in all collections", async () => {
+          setup({
+            clickBehavior,
+            dashboard: dashboardInPersonalCollection,
+            searchResults: [questionSearchResult],
+          });
+          userEvent.click(screen.getByRole("button", { name: "Search" }));
+          const typedText = "question";
+          userEvent.type(
+            screen.getByPlaceholderText("Search"),
+            `${typedText}{enter}`,
+          );
+
+          expect(
+            await screen.findByText(questionSearchResult.name),
+          ).toBeInTheDocument();
+
+          const call = fetchMock.lastCall("path:/api/search");
+          const urlObject = new URL(checkNotNull(call?.request?.url));
+          expect(urlObject.pathname).toEqual("/api/search");
+          expect(
+            fromEntriesWithMultipleValues(urlObject.searchParams.entries()),
+          ).toEqual({
+            models: ["card", "dataset"],
+            q: typedText,
+          });
+        });
+      });
+    });
+  });
+});
+
+function fromEntriesWithMultipleValues(
```

Comment:
The reason this one is needed is because previously I was comparing the `URLSearchParams` with:
https://github.com/metabase/metabase/blob/68788d775f5642046630ad34c98c4c5e0a5e9522/frontend/src/metabase/containers/AddToDashSelectDashModal/AddToDashSelectDashModal.unit.spec.tsx#L888-L891

This doesn't work with some of the tests in this file because it has multiple value for the same key i.e. `models=dataset&models=card`

using `Object.fromEntries` will just result in an object with the key `models` being either `dataset` or `card`.

This little util produces the value array, when it found multiple value for the same key.